lib/sms/sms_status_report.c

```diff
@@ -0,0 +1,386 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <string.h>
+#include <stdio.h>
+#include <zephyr.h>
+#include <modem/sms.h>
+#include <logging/log.h>
+
+#include "sms_status_report.h"
+#include "sms_internal.h"
+#include "parser.h"
+#include "string_conversion.h"
+
+LOG_MODULE_DECLARE(sms, CONFIG_SMS_LOG_LEVEL);
+
+/** @brief Maximum length of SMS address, i.e., phone number, in octets. */
+#define SMS_MAX_ADDRESS_LEN_OCTETS 10
+
+/** @brief Length of TP-Service-Centre-Time-Stamp field. */
+#define SCTS_FIELD_SIZE 7
+
+/** @brief Length of TP-Discharge-Time field. */
+#define DT_FIELD_SIZE 7
+
+/**
+ * @brief TP-Message-Type-Indicator in 3GPP TS 23.040 chapter 9.2.3.1
+ */
+#define TP_MTI_SMS_DELIVERY         0x00
```

Comment:
These TP_MTI constants should be in some generic header file.
Probably some of the above also as they are redefined in sms_deliver.c.

Reply:
Closing this comment as SMS library changes have been removed from this PR.